bus-mapping/src/circuit_input_builder/execution.rs

```diff
@@ -86,6 +86,45 @@ impl ExecStep {
             Some(ExecError::OutOfGas(_) | ExecError::StackOverflow | ExecError::StackUnderflow)
         )
     }
+
+    /// Try get opcode, if possible
+    pub fn opcode(&self) -> Option<OpcodeId> {
+        match self.exec_state {
+            ExecState::Op(op) => Some(op),
+            _ => None,
+        }
+    }
+
+    /// get rw index
+    pub fn rw_index(&self, index: usize) -> OperationRef {
+        self.bus_mapping_instance[index]
+    }
+
+    /// Get the size of read and writes
+    pub fn rw_indices_len(&self) -> usize {
+        self.bus_mapping_instance.len()
+    }
+
+    /// Get stack pointer
+    pub fn stack_pointer(&self) -> u64 {
+        1024 - self.stack_size as u64
+    }
+
+    /// The memory size in word **before** this step
+    pub fn memory_word_size(&self) -> u64 {
+        let n_bytes_word = 32u64;
+        let memory_size = self.memory_size as u64;
+        // EVM always pads the memory size to word size
+        // https://github.com/ethereum/go-ethereum/blob/master/core/vm/interpreter.go#L212-L216
```

Comment:
```suggestion
        // https://github.com/ethereum/go-ethereum/blob/a340721aa909ea4b541ffd1ea5e9c7bd441ff769/core/vm/interpreter.go#L201-L205
```
Updated link: the original was using a reference to `master` which changes over time, and it had already changed, so using a commit will make it stable over time.

Reply:
Good call!